File: src/Creational/Builder/InsurancePolicy.cs
namespace DesignPatterns.Creational.Builder;

public class InsurancePolicy
{
    public string PolicyType { get; set; }
    public double Premium { get; set; }
    public double CoverAmount { get; set; }

    public override string ToString()
    {
        return $"Policy Type: {PolicyType}, Premium: {Premium:C0}, Cover Amount: {CoverAmount:C0}";
    }
}

File: ts/src/creational/builder/InsurancePolicy.ts
export class InsurancePolicy {
    policyType?: string;
    premium?: number;
    coverAmount?: number;

    toString(): string {
        const premiumStr = this.premium !== undefined
            ? this.premium.toLocaleString('en-US', { style: 'currency', currency: 'USD', maximumFractionDigits: 0 })
            : '';
        const coverAmountStr = this.coverAmount !== undefined
            ? this.coverAmount.toLocaleString('en-US', { style: 'currency', currency: 'USD', maximumFractionDigits: 0 })
            : '';
        return `Policy Type: ${this.policyType}, Premium: ${premiumStr}, Cover Amount: ${coverAmountStr}`;
    }
}
